DOL.WHD.Section14c.Web/src/modules/components/resetPasswordForm/resetPasswordFormController.test.js

```diff
@@ -2,19 +2,201 @@ describe('resetPasswordFormController', function() {
 
     beforeEach(module('14c'));
 
-    beforeEach(inject(function ($rootScope, $controller) {
+    beforeEach(inject(function ($rootScope, $controller, _$q_, $location, apiService, stateService) {
         scope = $rootScope.$new();
+        $q = _$q_;
+        mockApiService = apiService;
+        mockStateService = { user: {} };
+        nockLocation = $location;
 
         resetPasswordFormController = function() {
             return $controller('resetPasswordFormController', {
-                '$scope': scope, 
-                'navService': mockNavService,
-                '$route': route
+                '$scope': scope,
+                'apiService': mockApiService,
+                'stateService': mockStateService,
+                '$location': nockLocation
             });
         };
+
+        resetPassword = $q.defer();
+        spyOn(mockApiService, 'resetPassword').and.returnValue(resetPassword.promise);
+        
+        verifyResetPassword = $q.defer();
+        spyOn(mockApiService, 'verifyResetPassword').and.returnValue(verifyResetPassword.promise);
+
     }));
 
-    it('invoke controller', function() {
+    it('reset passwork onSubmitClick success should show success message', function() {
         var controller = resetPasswordFormController();
-    });
+        scope.onSubmitClick();
+        resetPassword.resolve({}); 
+        scope.$apply();
+
+        expect(controller.forgotPasswordSuccess).toBe(true);
+    });    
+
+    it('reset passwork onSubmitClick error should show error message', function() {
+        var controller = resetPasswordFormController();
+        scope.onSubmitClick();
+        resetPassword.reject({}); 
+        scope.$apply();
+
+        expect(controller.forgotPasswordError).toBe(true);
+    });      
+
+    it('reset passwork onSubmitClick error should show error message, log details', function() {
+        var controller = resetPasswordFormController();
+        scope.onSubmitClick();
+        resetPassword.reject({ data: { error: "test" }}); 
+        scope.$apply();
+
+        expect(controller.forgotPasswordError).toBe(true);
```

Comment:
removed (again this was for branch coverage) https://github.com/AppliedIS/dol-whd-14c/commit/90f8d301128463df2a233ff2db7dfe4da358dba6